io/io/src/TBufferMerger.cxx

```diff
@@ -92,37 +89,25 @@ void TBufferMerger::SetAutoSave(size_t size)
 
 void TBufferMerger::Merge()
 {
-   fBuffered = 0;
-   fMerger.PartialMerge();
-   fMerger.Reset();
-
-   if (fCallback)
-      fCallback();
-}
-
-void TBufferMerger::WriteOutputFile()
-{
-   std::unique_ptr<TBufferFile> buffer;
-
-   while (true) {
-      std::unique_lock<std::mutex> lock(fQueueMutex);
-      fDataAvailable.wait(lock, [this]() { return !this->fQueue.empty(); });
-
-      buffer.reset(fQueue.front());
-      fQueue.pop();
-      lock.unlock();
-
-      if (!buffer)
-         break;
-
-      fBuffered += buffer->BufferSize();
-      fMerger.AddAdoptFile(new TMemFile(fMerger.GetOutputFileName(), buffer->Buffer(), buffer->BufferSize(), "read"));
-
-      if (fBuffered > fAutoSave)
-         Merge();
+   if (fMergeMutex.try_lock()) {
```

Comment:
Since Merge is executed on the user threads and since the critical sections calls the TMemFile constructor which may take the ROOT global locks, isn't this introducing a potential dead-lock?

Reply:
`fMergeMutex.try_lock()` doesn't block if the lock is acquired by someone else, so I don't think that there can be a deadlock.

Reply:
True.  Thanks.